src/main/java/net/smartcosmos/cluster/userdetails/UserDetailsResource.java

```diff
@@ -18,66 +16,47 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import net.smartcosmos.cluster.userdetails.domain.AuthenticateRequest;
+import net.smartcosmos.cluster.userdetails.domain.UserAuthenticationProperties;
 import net.smartcosmos.cluster.userdetails.domain.UserDto;
 
 @RestController
 @Slf4j
 public class UserDetailsResource {
 
     @Autowired
-    PasswordEncoder passwordEncoder;
+    private PasswordEncoder passwordEncoder;
 
+    private UserAuthenticationProperties userAuthenticationProperties;
+    @Autowired
+    public UserDetailsResource(UserAuthenticationProperties userAuthenticationProperties) {
+        this.userAuthenticationProperties = userAuthenticationProperties;
+    }
     @RequestMapping(value = "authenticate", method = RequestMethod.POST)
     public UserDto authenticate(@RequestBody @Valid AuthenticateRequest authentication)
-                    throws UsernameNotFoundException, IOException {
+        throws UsernameNotFoundException, IOException {
 
         log.info("Requested information on username {} with {}", authentication.getName(),
-                authentication);
+                 authentication);
 
         String passwordHash = null;
-        if (StringUtils.isNotBlank(authentication.getCredentials())) {
-            String credentials = authentication.getCredentials();
-            if (!"password".equals(credentials)) {
+        if (StringUtils.equals(userAuthenticationProperties.getName(), authentication.getPrincipal())) {
+            if (!StringUtils.equals(userAuthenticationProperties.getPassword(), authentication.getCredentials())) {
                 log.error("Password incorrect.");
                 throw new BadCredentialsException("Invalid password");
             }
-            passwordHash = passwordEncoder.encode(credentials);
+            passwordHash = passwordEncoder.encode(authentication.getCredentials());
         }
 
-        final String accountUrn = "urn:account:uuid:53f452c2-5a01-44fd-9956-3ecff7c32b30";
+        final String tenantUrn = "urn:account:uuid:53f452c2-5a01-44fd-9956-3ecff7c32b30";
```

Comment:
I only changed the variable name, not the value.  I think this goes back a long way.